# Matrix Operations

In [6]:
# import libraries
import pandas as pd
import numpy as np

**What are Matrices?**

In simple terms, matrix is a two-dimensional array of numbers arranged in rows and columns.

Matrices are used to represent datasets, where each row corresponds to an observation or sample, and each column represents a feature or attribute of that sample.

In this project, I'll be using a movie dataset where each row represents a movie, and each column represents a feature.


In [7]:
# load data into dataframe
df = pd.read_csv('../datasets/Hydra-Movie-Scrape.csv')


In [8]:
# create movie feature matrix
movie_matrix = df[['Runtime', 'Rating']].to_numpy()

Now that we have our feature matrix, let's check its shape

In [9]:
movie_matrix.shape

(3886, 2)

It is a 3886 x 2 matrix, meaning it has 3886 rows and 2 columns. Our matrix is tall and skinny.

In [10]:
# extract movie titles
movie_titles = df['Title']


Let's now check out the Transpose of the matrix.

In [11]:
movie_transpose = movie_matrix.T

In [17]:
print(f"Original matrix: \n{movie_matrix}\n")

print(f"Transposed matrix: \n{movie_transpose}")

Original matrix: 
[[ 66.    7.4]
 [ 75.    7.9]
 [ 65.    6.8]
 ...
 [114.    6.3]
 [ 94.    5.2]
 [100.    5.1]]

Transposed matrix: 
[[ 66.   75.   65.  ... 114.   94.  100. ]
 [  7.4   7.9   6.8 ...   6.3   5.2   5.1]]
